CMakeLists.txt

```diff
@@ -0,0 +1,1178 @@
+  libc-top-half/musl/src/complex/conjl.c
+  libc-top-half/musl/src/complex/cpow.c
+  libc-top-half/musl/src/complex/cpowf.c
+  libc-top-half/musl/src/complex/cpowl.c
+  libc-top-half/musl/src/complex/cproj.c
+  libc-top-half/musl/src/complex/cprojf.c
+  libc-top-half/musl/src/complex/cprojl.c
+  libc-top-half/musl/src/complex/creall.c
+  libc-top-half/musl/src/complex/csin.c
+  libc-top-half/musl/src/complex/csinf.c
+  libc-top-half/musl/src/complex/csinh.c
+  libc-top-half/musl/src/complex/csinhf.c
+  libc-top-half/musl/src/complex/csinhl.c
+  libc-top-half/musl/src/complex/csinl.c
+  libc-top-half/musl/src/complex/csqrt.c
+  libc-top-half/musl/src/complex/csqrtf.c
+  libc-top-half/musl/src/complex/csqrtl.c
+  libc-top-half/musl/src/complex/ctan.c
+  libc-top-half/musl/src/complex/ctanf.c
+  libc-top-half/musl/src/complex/ctanh.c
+  libc-top-half/musl/src/complex/ctanhf.c
+  libc-top-half/musl/src/complex/ctanhl.c
+  libc-top-half/musl/src/complex/ctanl.c
+  libc-top-half/musl/src/crypt/crypt_blowfish.c
+  libc-top-half/musl/src/crypt/crypt.c
+  libc-top-half/musl/src/crypt/crypt_des.c
+  libc-top-half/musl/src/crypt/crypt_md5.c
+  libc-top-half/musl/src/crypt/crypt_r.c
+  libc-top-half/musl/src/crypt/crypt_sha256.c
+  libc-top-half/musl/src/crypt/crypt_sha512.c
+  libc-top-half/musl/src/crypt/encrypt.c
+  libc-top-half/sources/arc4random.c)
+target_include_directories(libc-top-half PRIVATE
+  libc-top-half/musl/src/include
+  libc-top-half/musl/src/internal
+  libc-top-half/musl/arch/wasm32
+  libc-top-half/musl/arch/generic
+  libc-top-half/headers/private)
+target_compile_options(libc-top-half PRIVATE
+  -Wno-parentheses
+  -Wno-shift-op-parentheses
+  -Wno-bitwise-op-parentheses
+  -Wno-logical-op-parentheses
+  -Wno-string-plus-int
+  -Wno-dangling-else
+  -Wno-unknown-pragmas)
+add_dependencies(libc-top-half sysroot-headers)
+
+target_link_libraries(c PRIVATE
+  libc-bottom-half
+  libc-top-half
+  dlmalloc)
+
+foreach(stub m rt pthread crypt util xnet resolve dl)
+  add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/sysroot/lib/lib${stub}.a
+    COMMAND ${CMAKE_AR} crs ${CMAKE_BINARY_DIR}/sysroot/lib/lib${stub}.a)
+  add_custom_target(${stub} ALL
+    DEPENDS ${CMAKE_BINARY_DIR}/sysroot/lib/lib${stub}.a)
+endforeach()
+
+add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/sysroot/lib/libc.imports
+  COMMAND
+    ${Python_EXECUTABLE} ${PROJECT_SOURCE_DIR}/tools/gen-imports.py
+    --nm ${CMAKE_NM}
+    --output ${CMAKE_BINARY_DIR}/sysroot/lib/libc.imports
+    $<TARGET_FILE:c>
+    ${CMAKE_BINARY_DIR}/sysroot/lib/crt1.o
+  DEPENDS
+    ${PROJECT_SOURCE_DIR}/tools/gen-imports.py)
+add_custom_target(libc.imports ALL
+  DEPENDS ${CMAKE_BINARY_DIR}/sysroot/lib/libc.imports)
+add_dependencies(libc.imports c crt1.o)
+
+add_test(NAME check-metadata
+  COMMAND ${CMAKE_COMMAND} -E compare_files --ignore-eol ${PROJECT_SOURCE_DIR}/expected/wasm32-wasi/predefined-macros.txt ${CMAKE_BINARY_DIR}/sysroot/share/predefined-macros.txt)
```

Comment:
This is less friendly than running `diff` because it doesn't show what the differences were. Could you add a rule which runs an actual `diff` command?